sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -517,25 +526,45 @@ private[spark] object DiagnosticsConfig {
 
 private object ItemWriteStrategy extends Enumeration {
   type ItemWriteStrategy = Value
-  val ItemOverwrite, ItemAppend, ItemDelete, ItemDeleteIfNotModified, ItemOverwriteIfNotModified = Value
+  val ItemOverwrite, ItemAppend, ItemDelete, ItemDeleteIfNotModified, ItemOverwriteIfNotModified, ItemPatch = Value
 }
 
+private object CosmosPatchOperationTypes extends Enumeration {
+  type CosmosPatchOperationTypes = Value
+
+  val None = Value("none")
+  val Add = Value("add")
+  val Set = Value("set")
+  val Replace = Value("replace")
+  val Remove = Value("remove")
+  val Increment = Value("increment")
+}
+
+private case class CosmosPatchColumnConfig(columnName: String,
+                                           operationType: CosmosPatchOperationTypes,
+                                           mappingPath: String)
+
+private case class CosmosPatchConfigs(columnConfigsMap: TrieMap[String, CosmosPatchColumnConfig],
+                                      filter: Option[String] = None)
+
 private case class CosmosWriteConfig(itemWriteStrategy: ItemWriteStrategy,
                                      maxRetryCount: Int,
                                      bulkEnabled: Boolean,
                                      bulkMaxPendingOperations: Option[Int] = None,
                                      pointMaxConcurrency: Option[Int] = None,
-                                     maxConcurrentCosmosPartitions: Option[Int] = None)
+                                     maxConcurrentCosmosPartitions: Option[Int] = None,
+                                     patchConfigs: Option[CosmosPatchConfigs] = None)
 
 private object CosmosWriteConfig {
+  private val DefaultMaxRetryCount = 10
+  private val DefaultPatchOperationType = CosmosPatchOperationTypes.Replace
```

Comment:
Curious to know why are we defaulting to `Replace`?

Reply:
just based on our understanding, replace might be used more often. So we have chosen replace as default